### PR TITLE
MAILET-115 Manage every attributes in FakeMail.Builder

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/matchers/BaseMatchersTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.james.transport.matchers.All;
+import org.apache.james.transport.matchers.RecipientIs;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.Matcher;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.junit.Before;
+
+public class BaseMatchersTest {
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/matchers/BaseMatchersTest.java]: not related?

Comment (mbaechler) [inline server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/matchers/BaseMatchersTest.java]: of course, don't know why it's there

Comment (quynhn) [inline server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/matchers/BaseMatchersTest.java]: In fact, it contains some common functions setting which should be used on some matcher testing classes like: AndTest, OrTest, XorTest ...

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/RecipientRewriteTableTest.java [hunk 2]
```diff
@@ -93,14 +97,21 @@ public void testAddressMapping() throws Exception {
...
     private Mail createMail(String[] recipients) throws MessagingException {
-        Mail mail = new FakeMail();
-        ArrayList<MailAddress> a = new ArrayList<MailAddress>(recipients.length);
-        for (String recipient : recipients) {
-            a.add(new MailAddress(recipient));
-        }
-        mail.setRecipients(a);
-        mail.setMessage(new MimeMessage(NO_SESSION));
-        return mail;
+        return FakeMail.builder()
```
Comment (quynhn) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/RecipientRewriteTableTest.java]: No, it belongs to this commit because of re-factoring. It should be completely removed at another PR by RRT re-factoring

File: mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java [hunk 3]
```diff
@@ -214,57 +268,58 @@ public void setMessage(MimeMessage message) {
...
     }
-    
+
     public void setMessageSize(long size) {
         this.size = size;
     }
-
-    public void setRemoteAddr(String remoteAddr) {
-        this.remoteAddr = remoteAddr;
-    }
```
Comment (rouazana) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: you need to keep this, no?

Comment (aduprat) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: No

Comment (rouazana) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: strange because it is kept here: https://github.com/linagora/james-project/pull/402/commits/c75b72847f6754f293679471d07b7437f5613893

Comment (aduprat) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: Thanks, you are right !
I definitivly hate this Mail interface.

Comment (aduprat) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: Sorry, i misread the link, it's not used now and it's also not part of the Mail interface.
We can remove it.

Comment (rouazana) [inline mailet/base/src/test/java/org/apache/mailet/base/test/FakeMail.java]: OK if you say so...

**Comment by aduprat** [comment]
Rebased on #422
